flexget/plugins/clients/flood.py

```diff
@@ -0,0 +1,442 @@
+    }
+
+    def __init__(self) -> None:
+        self.flood = FloodClient()
+
+    def generate_entry(self, config: dict, torrent: dict) -> Entry:
+        """
+        Generate a task entry from a torrent in Flood.
+        """
+
+        entry = Entry(url='', title=torrent['name'])
+
+        for key, value in torrent.items():
+            entry['flood_' + key.lower()] = value
+
+        return entry
+
+    def on_task_input(self, task: Task, config: dict):
+        if self.flood.is_jwt_expired():
+            logger.debug("JWT token is expired. Re-authenticating with Flood.")
+            self.flood.authenticate(config)
+
+        entries = []
+        torrents = self.flood.list_torrents(config)
+
+        for hash, torrent in torrents.items():
+            entry = self.generate_entry(config, torrent)
+
+            if entry:
+                entries.append(entry)
+
+        return entries
+
+
+class OutputFlood:
+    """Add, remove, delete, start, or stop torrents in Flood.
+
+    Example:
+        flood:
+            url: http://localhost:3000
+            username: username
+            password: password
+            path: /downloads/Flexget
+            tags: Flexget
+            action: add
+    """
+
+    schema = {
+        'type': 'object',
+        'properties': {
+            'url': {'type': 'string'},
+            'username': {'type': 'string'},
+            'password': {'type': 'string'},
+            'action': {'type': 'string', 'enum': ['add', 'remove', 'delete', 'start', 'stop']},
+            'path': {'type': 'string'},
+            'tags': {'type': 'array', 'items': {'type': 'string'}},
+        },
+        'additionalProperties': False,
+        'required': ['url', 'action'],
+    }
+
+    def __init__(self) -> None:
+        self.flood = FloodClient()
+
+    def add_entry(self, config: dict, task: Task, entry: Entry) -> None:
+        """
+        Add an entry to Flood.
+        Will only attempt to add torrents if the JWT token is not expired or unset.
+        """
+
+        if self.flood.is_jwt_expired():
+            raise PluginError("Not authenticated with Flood.")
+
+        destination: str = entry.render(entry.get('path', ''), config.get('path', ''))
+        # Combines the tags from the entry and the config.
+        # The tags from the entry will be prioritized.
+        tags: list = entry.get('tags', []) + [
+            tag for tag in config.get('tags', []) if not tag in entry.get('tags', [])
+        ]
+        tags = [entry.render(tag) for tag in tags]
+        start: bool = entry.get('start', config.get('start', True))
+
+        if task.manager.options.test:
+            logger.info('Would add {} to Flood with options:', entry['title'])
+
+            if entry.get('file'):
+                logger.info('  File: {}', entry['file'])
+            elif entry.get('url'):
+                logger.info('  Url: {}', entry['url'])
+            else:
+                logger.info('  No URL or File found.')
```

Comment:
I'd consider extracting this to a helper method to reduce nesting.

Reply:
I decided to keep the test option logging as it was given that it's so small and the logic being quite simple.